pkg/query/api/v1.go

```diff
@@ -288,18 +290,24 @@ func (api *API) query(r *http.Request) (interface{}, []error, *ApiError) {
 		warnmtx.Unlock()
 	}
 
+	raw := r.FormValue("query")
+	level.Debug(api.logger).Log("query", raw)
+
 	// We are starting promQL tracing span here, because we have no control over promQL code.
 	span, ctx := tracing.StartSpan(r.Context(), "promql_instant_query")
+	span.SetTag("query", raw)
 	defer span.Finish()
 
 	begin := api.now()
-	qry, err := api.queryEngine.NewInstantQuery(api.queryableCreate(enableDedup, 0, enablePartialResponse, warningReporter), r.FormValue("query"), ts)
+	qry, err := api.queryEngine.NewInstantQuery(api.queryableCreate(enableDedup, 0, enablePartialResponse, warningReporter), raw, ts)
 	if err != nil {
+		level.Debug(api.logger).Log("query", raw, "error", errorBadData)
 		return nil, nil, &ApiError{errorBadData, err}
 	}
 
 	res := qry.Exec(ctx)
 	if res.Err != nil {
+		level.Debug(api.logger).Log("query", raw, "error", res.Err.Error(), "duration", time.Since(begin).Seconds())
```

Comment:
You already have the query printed above, so maybe those log lines on errors (debug) are enough?